contracts/nft/nft-pre-designed-collection.ral

```diff
@@ -3,31 +3,78 @@
 Contract NFTPreDesignedCollection(
     enumerableNftTemplateId: ByteVec,
     collectionUri: ByteVec,
-    tokenBaseUri: ByteVec,
     collectionOwner: Address,
+    tokenBaseUri: ByteVec,
+    maxSupply: U256,
+    mintPrice: U256,
     mut totalSupply: U256
-) extends NFTCollectionBase(collectionUri, totalSupply) {
-    @using(preapprovedAssets = true, updateFields = true)
-    pub fn mint() -> (ByteVec) {
+) {
+    enum ErrorCodes {
+        IncorrectTokenIndex = 0
+        NFTNotFound = 1
+        TokenOwnerAllowedOnly = 2
+    }
+
+    @using(preapprovedAssets = true, updateFields = true, assetsInContract = true)
+    pub fn mint(index: U256) -> (ByteVec) {
+        checkCaller!(index < maxSupply, ErrorCodes.IncorrectTokenIndex)
```

Comment:
How about using `assert!` and adding `checkExternalCaller = false` annotation? I feel that using `checkCaller` here might be confusing because anyone can call this function.

Reply:
Sounds good, fixed :+1: